packages/mui-base/src/useAutocomplete/useAutocomplete.js

```diff
@@ -1167,7 +1168,7 @@ export function useAutocomplete(props) {
       const disabled = getOptionDisabled ? getOptionDisabled(option) : false;
 
       return {
-        key: getOptionLabel(option),
+        key: getOptionKey(option) ?? getOptionLabel(option),
```

Comment:
To prevent potentially breaking existing applications, let's not assign a default value to `getOptionKey` and when it's undefined, call `getOptionLabel`, as before:

```suggestion
        key: getOptionKey?.(option) ?? getOptionLabel(option),
```